nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviService.java

```diff
@@ -47,8 +47,11 @@ public static NviService defaultNviService() {
     //TODO: Remove JacocoGenerated when other if/else cases are implemented
     @JacocoGenerated
     public Optional<CandidateWithIdentifier> upsertCandidate(CandidateEvaluatedMessage evaluatedCandidate) {
-        if (isNotExistingCandidate(evaluatedCandidate) && isNviCandidate(evaluatedCandidate)) {
+        if (!isExistingCandidate(evaluatedCandidate) && isNviCandidate(evaluatedCandidate)) {
             return Optional.of(createCandidate(evaluatedCandidate));
+        } else if (isExistingCandidate(evaluatedCandidate) && isNviCandidate(evaluatedCandidate)) {
+            var existing = findByPublicationId(evaluatedCandidate.candidateDetails().publicationId()).orElseThrow();
+            return Optional.of(updateCandidate(existing.identifier(), evaluatedCandidate));
         }
         return Optional.empty();
```

Comment:
Can be written in more attempt() way, but it is up to you :)
```
return doesNotExist(evaluatedCandidate)
           ? Optional.of(createCandidate(evaluatedCandidate))
           : Optional.of(evaluatedCandidate.candidateDetails().publicationId())
                 .map(this::findByPublicationId)
                 .map(Optional::orElseThrow)
                 .map(CandidateWithIdentifier::identifier)
                 .map(id -> updateCandidate(id, evaluatedCandidate))
                 .or(Optional::empty);
```

Reply:
Thanks for the input and the code example. It would be better if the code was finished, but now we are missing some cases, and I think it will be more efficient to refactor it after we have accounted for all cases